cmake/IgnUtils.cmake

```diff
@@ -1804,3 +1804,172 @@ macro(_ign_cmake_parse_arguments prefix options oneValueArgs multiValueArgs)
   endif()
 
 endmacro()
+ 
+#####GSOC##########################################
+
+#ign_add_resources(path_to_resources)
+#Installs the folder and provides it's path to ign_environment_hook()
+macro(ign_add_resources path_to_resources)
+
+  install(DIRECTORY
+  ${path_to_resources}
+  DESTINATION share/${PROJECT_NAME})
+
+  list(APPEND resources_path ${path_to_resources})
+
+endmacro() 
+#####################################################
+#ign_environment_hook()
+#Get's path from other macros and creates hooks to export them
+#Currently it creates colcon.pkg along with hooks.dsv for exporting them
+#Mechanisms for plain cmake packages is under development 
+macro(ign_environment_hook)
+  list(REMOVE_DUPLICATES resources_path)
+  list(REMOVE_DUPLICATES plugins_path)
+
+
+  foreach(resource_path ${resources_path})
+    file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in
+    "prepend-non-duplicate;IGN_GAZEBO_RESOURCE_PATH;@CMAKE_INSTALL_PREFIX@/share/@PROJECT_NAME@/${resource_path}\n")
+  endforeach()
+  foreach(plugin_path ${plugins_path})
+    file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in 
+    "prepend-non-duplicate;IGN_GAZEBO_SYSTEM_PLUGIN_PATH;@CMAKE_INSTALL_PREFIX@/${plugin_path}\n")
+  endforeach()
+  file(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/colcon.pkg "{\n  \"hooks\": [\"share/my_package/hooks/hook.dsv\"]\n}")
+  file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in 
+  "prepend-non-duplicate;LD_LIBRARY_PATH;@CMAKE_INSTALL_PREFIX@/lib\n")
+  
+
+  configure_file(
+    "hooks/hook.dsv.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/hooks/hook.dsv" @ONLY
+  )
+  install(DIRECTORY
+  ${CMAKE_CURRENT_BINARY_DIR}/hooks
+  DESTINATION share/${PROJECT_NAME})
+
+endmacro()
+#####################################################
+#ign_add_plugins(path_to_plugin)
+#Installs all the plugins inside the folder using various arguments from the user
+#Also provides the path to ign_environment_hook()
+#For adding target_link_libraries or target_include_directories for a specific plugin 
+#set ${PLUGIN_NAME}_(PUBLIC/PRIVATE)_(LIBRARIES/DIRECTORIES) as required plugins or directories in your package
+
+macro(ign_add_plugins path_to_plugin )
+
+  file(GLOB source_list CONFIGURE_DEPENDS
+      "${path_to_plugin}/*.cc"
```

Comment:
Again this assumes certain folder structure. Its fine to keep it this way for early prototyping as we iterate on the tool but, it would be better if we had them directly adding files to this call. Further what if someone wants to add dependencies.